tools/codegen/api/cpp.py

```diff
@@ -23,9 +23,9 @@
 # BTW: policy on name collisions: we try not to have types with
 # collisions, but functions are fair game to collide
 
-def name(func: FunctionSchema) -> str:
+def name(func: FunctionSchema, *, use_suffix_for_out_overloads: bool = True) -> str:
```

Comment:
Seems like this should be a property (maybe synthetic) of `FunctionSchema`

Reply:
It can't be part of FunctionSchema because we need to get both the faithful (`add`) and the unfaithful (`add_out`) name for a single FunctionSchema, there is no FunctionSchema variation. What this does suggest, though, was I was wrong to have individual functions for extracting name here, and we should go full on with CppSignature (which will avoid this very strange usage edge).